src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -426,6 +426,57 @@ def count_performance_issues(trace_id: str, params: Mapping[str, str]) -> int:
     return cast(int, count["data"][0].get("total_groups", 0))
 
 
+@sentry_sdk.tracing.trace
+def update_params_with_trace_timestamp_projects(
+    trace_id: str,
+    params: Mapping[str, str],
+) -> None:
+    query_metadata = options.get("performance.traces.query_timestamp_projects")
+    sentry_sdk.set_tag("trace_view.queried_timestamp_projects", query_metadata)
+    if not query_metadata:
+        return
+
+    metadata_query = QueryBuilder(
+        Dataset.Discover,
+        params,
+        query=f"trace:{trace_id}",
+        selected_columns=[
+            "min(timestamp)",
+            "max(timestamp)",
+            "project.id",
+        ],
+    )
+    results = metadata_query.run_query(Referrer.API_TRACE_VIEW_GET_TIMESTAMP_PROJECTS.value)
+    results = metadata_query.process_results(results)
+    project_id_set = set()
+    min_timestamp = None
+    max_timestamp = None
+    for row in results["data"]:
```

Comment:
any chance this can return empty?

Reply:
Good catch, i'll add something so if the results are empty we don't touch params